core/store.go

```diff
@@ -238,6 +240,37 @@ func Rename(sourceKey, destKey string) bool {
 	}, WithStoreLock(), WithKeypoolLock())
 }
 
+// TODO, Key expiry should be handled for the RandomKey command
+// with the below implementations key expiry checks will be difficult(in Redis expiration is checked with maxTries of 100)
+// this should be considered on future data structure refactoring
+func RandomKey() string {
+	randomKey := ""
+	withLocks(func() {
+		if len(keypool) == 0 {
+			return
+		}
+		idxRange := len(keypool) / 2
+		if idxRange > 1024 {
+			idxRange = 1024
+		}
+
+		index, err := rand.Int(rand.Reader, big.NewInt(int64(idxRange)))
+		if err != nil {
+			return
+		}
+		RandomIndex := int(index.Int64())
+		currentIndex := 0
+		for key := range keypool {
+			if currentIndex == RandomIndex {
+				randomKey = key
```

Comment:
You need to ensure that the key being returned has not expired

Reply:
@JyotinderSingh , for this expiration check , redis using a maxtries of 100.  With our current Map structure how to  do this efficiently?

Reply:
One option is to simply keep checking the next key until you encounter a valid key.
@soumya-codes thoughts?

Reply:
@JyotinderSingh I was thinking if we can leverage parts of SCAN capability to address RANDOMKEY effectively. wdyt?